### PR TITLE
[Mapping.NonLinear] Move DistanceMultiMapping in its own files

### PR DESCRIPTION
**Pull request by alxbilger**
It was initially defined in the same files than DistanceMapping

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/Mapping/NonLinear/src/sofa/component/mapping/nonlinear/DistanceMapping.h [hunk 1]
```diff
@@ -24,12 +24,12 @@
 #include <sofa/component/mapping/nonlinear/config.h>
 
 #include <sofa/core/Mapping.h>
-#include <sofa/core/MultiMapping.h>
 #include <sofa/linearalgebra/EigenSparseMatrix.h>
 #include <sofa/core/topology/BaseMeshTopology.h>
 #include <sofa/type/Mat.h>
 #include <sofa/type/Vec.h>
 #include <sofa/type/RGBAColor.h>
+#include <sofa/component/mapping/nonlinear/DistanceMultiMapping.h>
```
Comment (fredroy) [inline Sofa/Component/Mapping/NonLinear/src/sofa/component/mapping/nonlinear/DistanceMapping.h:34]: Maybe add a comment to remove this header in v23.12, as it was to keep a compat with the previous behavior (i.e being able to use DistanceMultiMapping by including DistanceMapping.h)

**Review by fredroy — APPROVED** [review]

File: Sofa/Component/Mapping/NonLinear/src/sofa/component/mapping/nonlinear/DistanceMapping.h (hunk 1 above)
Comment (alxbilger) [reply Sofa/Component/Mapping/NonLinear/src/sofa/component/mapping/nonlinear/DistanceMapping.h:34]: I was thinking about it, but what does it change to remove it in v23.12 or now? In both cases, the user is not noticed (except with a compilation error). And I don't see how it is possible to warn the user if `DistanceMultiMapping` is used only via the include of `DistanceMapping`. What do you think?

Comment (fredroy) [reply Sofa/Component/Mapping/NonLinear/src/sofa/component/mapping/nonlinear/DistanceMapping.h:34]: Ah yes I totally agree with you on the fact it is quite impossible to warn the user, I was suggesting to add a dev comment (as with `//` or `/* */` ) for the next time someone stumbled on this inclusion (and add a timeline for @hugtalbot when he will do the next deprecation round in the future 😁)

**Pull request merged**